File: Android/TRTCScenesDemo/trtcvoiceroomdemo/src/main/java/com/tencent/liteav/trtcvoiceroom/model/impl/room/ITXRoomServiceDelegate.java
package com.tencent.liteav.trtcvoiceroom.model.impl.room;


import com.tencent.liteav.trtcvoiceroom.model.impl.base.TXRoomInfo;
import com.tencent.liteav.trtcvoiceroom.model.impl.base.TXSeatInfo;
import com.tencent.liteav.trtcvoiceroom.model.impl.base.TXUserInfo;

import java.util.List;

public interface ITXRoomServiceDelegate {
    void onRoomDestroy(String roomId);

    void onRoomRecvRoomTextMsg(String roomId, String message, TXUserInfo userInfo);

    void onRoomRecvRoomCustomMsg(String roomId, String cmd, String message, TXUserInfo userInfo);

    void onRoomInfoChange(TXRoomInfo TXRoomInfo);

    void onSeatInfoListChange(List<TXSeatInfo> TXSeatInfoList);

    void onRoomAudienceEnter(TXUserInfo userInfo);

    void onRoomAudienceLeave(TXUserInfo userInfo);

    void onSeatTake(int index, TXUserInfo userInfo);

    void onSeatClose(int index, boolean isClose);

    void onSeatLeave(int index, TXUserInfo userInfo);

    void onSeatMute(int index, boolean mute);

    void onReceiveNewInvitation(String id, String inviter, String cmd, String content);

    void onInviteeAccepted(String id, String invitee);

    void onInviteeRejected(String id, String invitee);

    void onInvitationCancelled(String id, String inviter);
}

File: iOS/TRTCScenesDemo/TXLiteAVDemo/TRTCVoiceRoomDemo/model/Impl/room/ITXRoomServiceDelegate.swift
//
//  ITXRoomServiceDelegate.swift
//  TRTCVoiceRoomDemo
//
//  Created by abyyxwang on 2020/6/9.
//  Copyright © 2020 tencent. All rights reserved.
//

import Foundation

public protocol ITXRoomServiceDelegate: class {
    func onRoomDestroy(roomID: String)
    func onRoomRecvRoomTextMsg(roomID: String, message: String, userInfo: TXUserInfo)
    func onRoomRecvRoomCustomMsg(roomID: String, cmd: String, message: String, userInfo: TXUserInfo)
    func onRoomInfoChange(roomInfo: TXRoomInfo)
    func onSeatInfoListChange(seatInfoList: [TXSeatInfo])
    func onRoomAudienceEnter(userInfo: TXUserInfo)
    func onRoomAudienceLeave(userInfo: TXUserInfo)
    func onSeatTake(index: Int, userInfo: TXUserInfo)
    func onSeatClose(index: Int, isClose: Bool)
    func onSeatLeave(index: Int, userInfo: TXUserInfo)
    func onSeatMute(index: Int, mute: Bool)
    func onReceiveNewInvitation(identifier: String, inviter: String, cmd: String, content: String)
    func onInviteeAccepted(identifier: String, invitee: String)
    func onInviteeRejected(identifier: String, invitee: String)
    func onInviteeCancelled(identifier: String, invitee: String)
}
